libs/source/Source.mli

```diff
@@ -0,0 +1,60 @@
+(* Cooper Pierce
+ *
+ * Copyright (c) Semgrep Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+
+(** Sources of content which may be consumed by Semgrep.
+
+   A source is the {e user-relevant location} where content (e.g., source code)
+   originated. This should be used when storing a location which will later be
+   reported to a user.
+
+   While a source may imply that content can be obtained from a given location,
+   it should generally be stored alongside the content to be consumed. This is
+   so that it can be used principally for reporting where a finding originated
+   independently of getting the content, which allows de-coupling reporting
+   where a finding originated from obtaining the contents.
+
+   This is since when these uses are fused, it is more difficult to
```

Comment:
I don't understand the sentence.

Reply:
Can you elaborate? It is not immediately obvious to me what to expand on based on your comment.

My best guess is that it may not be clear what "these uses" refers to. It is the actions mentioned in the preceding paragraph, namely (1) reporting where a finding originated and (2) obtaining the contents (e.g., source code) of the object from which the finding may be generated.

Reply:
Reworded in 5fee815255b446b1e3360b63d94021332adf6f60.